src/test/java/mate/academy/security/CustomUserDetailsServiceTest.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+class CustomUserDetailsServiceTest {
+    private static final String EMAIL = "bob@i.ua";
+    private static final String PASSWORD = "111111111";
+    private static final String WRONG_EMAIL = "alice@i.ua";
+    private UserDetailsService userDetailsService;
+    private UserService userService;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        userDetailsService = new CustomUserDetailsService(userService);
+        user = new User();
+        user.setEmail(EMAIL);
+        user.setPassword(PASSWORD);
+        user.setRoles(Set.of(new Role(Role.RoleName.USER)));
+    }
+
+    @Test
+    void loadUserByUsername_Ok() {
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(user));
+
+        UserDetails actual = userDetailsService.loadUserByUsername(EMAIL);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(EMAIL, actual.getUsername());
+        Assertions.assertEquals(PASSWORD, actual.getPassword());
+    }
+
+    @Test
+    void loadUserByUsername_UserNotFound() {
+        Mockito.when(userService.findByEmail(WRONG_EMAIL)).thenReturn(Optional.of(user));
```

Comment:
The test case for user not found is using wrong return value. It should return an empty Optional instead of the user object.